### ISSUE TITLE
Ethernet header

### ISSUE THREAD

**Issue by reinisb** [nervous-inhuman/tplink-tapo-c200-re#9]
Hello! Is it possible to use the ethernet header to operate the camere using ethernet instead of wifi? In other words, it would be perfect if I could solder an ethernet cable to the on-board ethernet header pins and operate the camera through it instead of wifi.

**Comment by reinisb** [comment]
@nervous-inhuman do you happen to know?

**Comment by nervous-inhuman** [comment]
No, I have not tried to do so. But feel free to try, and maybe share your findings :)

**Comment by SKlein-1428** [comment]
I'd like to add that this is absolutely possible! I got all of my Version 1 C100s and C200s connected via eithernet using the pinout provided in this repo - you just have to make sure to choose the C300 as "which Camera do you have?" in the app, and choose Ethernet as the connection, and it should find it. The LED is solid green rather than solid Amber during setup.

Edit: I made the task easier and more versatile by using a picoblade connector spliced into an ethernet male-to-female to make a dirty alternative to the C300 connector after my first proof-of-concept attempt.

**Comment by reinisb** [comment]
@SKlein-1428 This is great! I'll make sure to try it out!

PS An end game would be for the ethernet cable to also carry power, I suppose I need to squeeze in a little POE board inside the C200.

**Comment by SKlein-1428** [comment]
@reinisb https://user-images.githubusercontent.com/4379661/101809993-8c0f5100-3b18-11eb-9080-01ac77437e04.jpg hopefully this works? But the pinout I used is the one from this repo anyway, in the readme.

The power supply being just a two pin molex means you should be able to easily hook up a small daughter board to regulate the PoE and make it the right voltage and amperages? I've not messed with PoE myself in general but that does sound like a fun/useful addition, and a great way to turn a fairly cheap camera into a smaller, more self-sustained package that isn't vulnerable to the recent 2.4ghz disconnect attacks.

**Comment by xdom** [comment]
Just tried with Tapo C110 (EU model ver. 2.0) with no luck. Camera recognizes Ethernet is connected, as LED lights up solid amber, but no RX packets received on router. I can't rule out soldering issue, or any other issue.

**Comment by lin-xianming** [comment]
@SKlein-1428 I cannot find C300 in the Tapo app and cannot find any information about the existence of C300. Do you mean another model in the C3xx series?

> Just tried with Tapo C110 (EU model ver. 2.0) with no luck. Camera recognizes Ethernet is connected, as LED lights up solid amber, but no RX packets received on router. I can't rule out soldering issue, or any other issue.

Based on the boot logs seen in this repo the eth interface normally does not have a mac address assigned to it. If firmware check fails then it gets assigned 00:00:23:34:45:66 and the camera goes into failsafe mode.

Furthermore it's inadvisable to simply splice a cable directly to the PHY for long term use. You need a magnetic transformer or a magjack with an integrated magnetic transformer.

**Comment by SKlein-1428** [comment]
@lin-xianming I meant any of the C3xx that have ethernet natively. I could have been more clear, I guess.

Explain what you mean about PHY and magnetic transformers?

**Comment by lin-xianming** [comment]
Spec requires a magnetic transformer between the PHY and cable for protection against damage and improved signalling. The PHY in this case exists in the RTL3903. You can see four traces going directly from the RTL3903 to the ETH header.

You can see the magnetic transformer on the board for the C320WS which comes with an ethernet jack.

![image](https://github.com/nervous-inhuman/tplink-tapo-c200-re/assets/108489988/65513b84-8e8a-4293-8ecb-93f62d4c595e)
